Products/ZenModel/migrate/addIndexesForDevice.py

```diff
@@ -0,0 +1,50 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import Migrate
+import logging
+from Products.Zuul.catalog.model_catalog_init import reindex_model_catalog
+from Products.Zuul.catalog.indexable import DeviceIndexable
+from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
+from Products import Zuul
+
+log = logging.getLogger("zen.migrate")
+
+class AddIndexesForDevice(Migrate.Step):
+    """
+    Adds new indexes for devices in scope of adding new router which gives a possibility
+    to pull data directly from SOLR without touching ZODB.
+    """
+
+    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION)
+
+    def cutover(self, dmd):
+
+        f = Zuul.getFacade('modelquery')
+        device = f.getDevices(limit=1, fields=["collector"])
+
+        if device[0][0].get('collector'):
```

Comment:
The check is wrong, it should be `if device[0][0].get('collector') is None`
You could use model catalog as well:
```
search_results = IModelCatalogTool(dmd).devices.search(limit=1, fields=["collector"])
if search_results.total > 0 and search_results.results.next().collector is None:
    reindex
```